Add Smithery CLI Installation Instructions

calclavia opened the pull request:
This PR adds installation instructions to automatically install Neon MCP Server for Claude Desktop using Smithery CLI. This makes it easier for users to install the MCP. Also, it adds a badge to show the number of installations from Smithery.

pffigueiredo commented:
@calclavia, thanks for proactively adding this integration 🙏 

We will accept this PR once these are fixed on the CLI:

- the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
- the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable

**smithery CLI** ❌ 
```
{
  "mcpServers": {
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
    }
  }
}
```



**neon correct config** ✅ 
```
{
  "mcpServers": {
    "neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        <NEON_API_KEY>
      ]
    }
  }
}
```

calclavia commented:
> @calclavia, thanks for proactively adding this integration 🙏
> 
> We will accept this PR once these are fixed on the CLI:
> 
> * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> 
> **smithery CLI** ❌
> 
> ```
> {
>   "mcpServers": {
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
>     }
>   }
> }
> ```
> 
> **neon correct config** ✅
> 
> ```
> {
>   "mcpServers": {
>     "neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         <NEON_API_KEY>
>       ]
>     }
>   }
> }
> ```

Thanks for identifying the typo! Just fixed on our end.

![Screenshot 2024-12-16 at 7 35 12 PM](https://github.com/user-attachments/assets/8a8266c8-99d3-4147-8dc7-2ac3400460b4)

claude_desktop_config.json:
```
...
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "test"
      ]
    }
```

pffigueiredo commented:
i

> > @calclavia, thanks for proactively adding this integration 🙏
> > We will accept this PR once these are fixed on the CLI:
> > 
> > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > 
> > **smithery CLI** ❌
> > ```
> > {
> >   "mcpServers": {
> >     "@neondatabase-mcp-server-neon": {
> >       "command": "npx",
> >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> >     }
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **neon correct config** ✅
> > ```
> > {
> >   "mcpServers": {
> >     "neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         <NEON_API_KEY>
> >       ]
> >     }
> >   }
> > }
> > ```
> 
> Thanks for identifying the typo! Just fixed on our end.
> 
> ![Screenshot 2024-12-16 at 7 35 12 PM](https://private-user-images.githubusercontent.com/1828968/396079291-8a8266c8-99d3-4147-8dc7-2ac3400460b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTA2NDYsIm5iZiI6MTczNDM1MDM0NiwicGF0aCI6Ii8xODI4OTY4LzM5NjA3OTI5MS04YTgyNjZjOC05OWQzLTQxNDctOGRjNy0yYWMzNDAwNDYwYjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTE1OTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWFlMTYyZmM0ZGI4ODY4MzU1MWVhNWRlY2ZmMGZkYTIwNTE4NDlkYWE4YmQ4YWQxOTBlZWMwM2E0MDFmZGU4NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xSoOGsNZ6mCCMdZAid7rjPAkfkmckJ1nCrqAnZDYzAw)
> 
> claude_desktop_config.json:
> 
> ```
> ...
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "test"
>       ]
>     }
> ```

I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀 

```
{
  "mcpServers": {
    "@neondatabase-mcp-server-neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "my-key"
      ]
    }
  }
}
```

The rest is looking great thou 💪

calclavia commented:
> i
> 
> > > @calclavia, thanks for proactively adding this integration 🙏
> > > We will accept this PR once these are fixed on the CLI:
> > > 
> > > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > > 
> > > **smithery CLI** ❌
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "@neondatabase-mcp-server-neon": {
> > >       "command": "npx",
> > >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> > >     }
> > >   }
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > **neon correct config** ✅
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "neon": {
> > >       "command": "npx",
> > >       "args": [
> > >         "-y",
> > >         "@neondatabase/mcp-server-neon",
> > >         "start",
> > >         <NEON_API_KEY>
> > >       ]
> > >     }
> > >   }
> > > }
> > > ```
> > 
> > 
> > Thanks for identifying the typo! Just fixed on our end.
> > ![Screenshot 2024-12-16 at 7 35 12 PM](https://private-user-images.githubusercontent.com/1828968/396079291-8a8266c8-99d3-4147-8dc7-2ac3400460b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTA2NDYsIm5iZiI6MTczNDM1MDM0NiwicGF0aCI6Ii8xODI4OTY4LzM5NjA3OTI5MS04YTgyNjZjOC05OWQzLTQxNDctOGRjNy0yYWMzNDAwNDYwYjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTE1OTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWFlMTYyZmM0ZGI4ODY4MzU1MWVhNWRlY2ZmMGZkYTIwNTE4NDlkYWE4YmQ4YWQxOTBlZWMwM2E0MDFmZGU4NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xSoOGsNZ6mCCMdZAid7rjPAkfkmckJ1nCrqAnZDYzAw)
> > claude_desktop_config.json:
> > ```
> > ...
> >     "@neondatabase-mcp-server-neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         "test"
> >       ]
> >     }
> > ```
> 
> I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀
> 
> ```
> {
>   "mcpServers": {
>     "@neondatabase-mcp-server-neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "my-key"
>       ]
>     }
>   }
> }
> ```
> 
> The rest is looking great thou 💪

Ah that's mainly because our IDs were namespaced based on your npm package ID. Claude doesn't allow slashes in the name.

Anyway, I switched it to just `neon`. It's cleaner. 😊

calclavia commented:
> > > i
> > > > > @calclavia, thanks for proactively adding this integration 🙏
> > > > > We will accept this PR once these are fixed on the CLI:
> > > > > 
> > > > > * the name of the server is `neon` and not `@neondatabase-mcp-server-neon`
> > > > > * the NEON_API_KEY is set as the last argument, and not through instead of an `env` variable
> > > > > 
> > > > > **smithery CLI** ❌
> > > > > ```
> > > > > {
> > > > >   "mcpServers": {
> > > > >     "@neondatabase-mcp-server-neon": {
> > > > >       "command": "npx",
> > > > >       "args": ["-y", "@neondatabase/mcp-server-neon", "start"]
> > > > >     }
> > > > >   }
> > > > > }
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > **neon correct config** ✅
> > > > > ```
> > > > > {
> > > > >   "mcpServers": {
> > > > >     "neon": {
> > > > >       "command": "npx",
> > > > >       "args": [
> > > > >         "-y",
> > > > >         "@neondatabase/mcp-server-neon",
> > > > >         "start",
> > > > >         <NEON_API_KEY>
> > > > >       ]
> > > > >     }
> > > > >   }
> > > > > }
> > > > > ```
> > > > 
> > > > 
> > > > Thanks for identifying the typo! Just fixed on our end.
> > > > ![Screenshot 2024-12-16 at 7 35 12 PM](https://private-user-images.githubusercontent.com/1828968/396079291-8a8266c8-99d3-4147-8dc7-2ac3400460b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzNTA2NDYsIm5iZiI6MTczNDM1MDM0NiwicGF0aCI6Ii8xODI4OTY4LzM5NjA3OTI5MS04YTgyNjZjOC05OWQzLTQxNDctOGRjNy0yYWMzNDAwNDYwYjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTZUMTE1OTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWFlMTYyZmM0ZGI4ODY4MzU1MWVhNWRlY2ZmMGZkYTIwNTE4NDlkYWE4YmQ4YWQxOTBlZWMwM2E0MDFmZGU4NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xSoOGsNZ6mCCMdZAid7rjPAkfkmckJ1nCrqAnZDYzAw)
> > > > claude_desktop_config.json:
> > > > ```
> > > > ...
> > > >     "@neondatabase-mcp-server-neon": {
> > > >       "command": "npx",
> > > >       "args": [
> > > >         "-y",
> > > >         "@neondatabase/mcp-server-neon",
> > > >         "start",
> > > >         "test"
> > > >       ]
> > > >     }
> > > > ```
> > > 
> > > 
> > > I just ran the `npx @smithery/cli install @neondatabase/mcp-server-neon --client claude` command and the name of the server is still being written as `@neondatabase-mcp-server-neon` instead of just `neon` 👀
> > > ```
> > > {
> > >   "mcpServers": {
> > >     "@neondatabase-mcp-server-neon": {
> > >       "command": "npx",
> > >       "args": [
> > >         "-y",
> > >         "@neondatabase/mcp-server-neon",
> > >         "start",
> > >         "my-key"
> > >       ]
> > >     }
> > >   }
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The rest is looking great thou 💪
> > 
> > 
> > Ah that's mainly because our IDs were namespaced based on your npm package ID. Claude doesn't allow slashes in the name.
> > Anyway, I switched it to just `neon`. It's cleaner. 😊
> 
> Ah, I see!
> 
> Ideally it would just be `neon`, but it that doesn't work for you at all, I think we are OK with having "@neondatabase-mcp-server-neon" as the `key` of the MCP Server instead, this last update doesn't work because `neon` isn't a pacakge that `npx` can use 😓
> 
> So let's do one of these:
> 
> 1- Rollback to the previous version and just use "@neondatabase-mcp-server-neon" as the `key` in the `mcpServers` JSON 2- Use "neon" as the `key`, but keep "@neondatabase/mcp-server-neon" as the name of the package
> 
> Option 2 is preferred, but we can go with option 1, if that's too hard to get right on your side 👍
> 
> Sorry for all the back and forth, didn't understand that the name of the server was coupled with the name of the package!

Oh, the ID of the installation on Smithery is not coupled with the NPM package. I just named it based on how you defined it in the package, so no changes are needed and it should just work now.

pffigueiredo commented:
> @neondatabase/mcp-server-neon

So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:

```json
{
  "mcpServers": {
    "neon": {
      "command": "npx",
      "args": ["-y", "neon", "start", "API_KEY"]
    }
  }
}

```

And this is what it should be:
```json
 "neon": {
      "command": "npx",
      "args": [
        "-y",
        "@neondatabase/mcp-server-neon",
        "start",
        "API_KEY"
      ]
    }
```


The only difference right now is just the name of the package that `npx` should pick up

calclavia commented:
> > @neondatabase/mcp-server-neon
> 
> So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:
> 
> ```json
> {
>   "mcpServers": {
>     "neon": {
>       "command": "npx",
>       "args": ["-y", "neon", "start", "API_KEY"]
>     }
>   }
> }
> ```
> 
> And this is what it should be:
> 
> ```json
>  "neon": {
>       "command": "npx",
>       "args": [
>         "-y",
>         "@neondatabase/mcp-server-neon",
>         "start",
>         "API_KEY"
>       ]
>     }
> ```
> 
> The only difference right now is just the name of the package that `npx` should pick up

Fixed this. Now it correctly sets `@neondatabase/mcp-server-neon` while using `neon` as the name

<img width="254" alt="Screenshot 2024-12-17 at 10 55 23 AM" src="https://github.com/user-attachments/assets/6da7dfda-fd41-4f31-b839-09add52aaef0" />

pffigueiredo commented:
> > > @neondatabase/mcp-server-neon
> > 
> > 
> > So, I think we got a bit confused here, this is the output of `npx -y @smithery/cli install neon --client claude`:
> > ```json
> > {
> >   "mcpServers": {
> >     "neon": {
> >       "command": "npx",
> >       "args": ["-y", "neon", "start", "API_KEY"]
> >     }
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And this is what it should be:
> > ```json
> >  "neon": {
> >       "command": "npx",
> >       "args": [
> >         "-y",
> >         "@neondatabase/mcp-server-neon",
> >         "start",
> >         "API_KEY"
> >       ]
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The only difference right now is just the name of the package that `npx` should pick up
> 
> Fixed this. Now it correctly sets `@neondatabase/mcp-server-neon` while using `neon` as the name
> 
> <img alt="Screenshot 2024-12-17 at 10 55 23 AM" width="254" src="https://private-user-images.githubusercontent.com/1828968/396333705-6da7dfda-fd41-4f31-b839-09add52aaef0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0MzE2ODUsIm5iZiI6MTczNDQzMTM4NSwicGF0aCI6Ii8xODI4OTY4LzM5NjMzMzcwNS02ZGE3ZGZkYS1mZDQxLTRmMzEtYjgzOS0wOWFkZDUyYWFlZjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTdUMTAyOTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2M2Yzc2MDNmYTJhNDNhNDkwODg0NWMzNzVkNzQ1MTgzMzI0ZjkyOThiOTU1NWU1M2EyNWY3MjExMjAzMWE2MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9pYtTAf0xu6voRrNQ9Q7wIIQg51EooEIdTzrrAWv3Cw">

Awesome, I just end-to-end tested and it's working great  💪 

Thanks a lot for getting this though 🚀

pffigueiredo commented on README.md:
````diff
@@ -10,6 +12,14 @@ This lets you use Claude Desktop, or any MCP Client, to use natural language to
+To install Neon MCP Server for Claude Desktop automatically via [Smithery](https://smithery.ai/protocol/neon):
+
+```bash
+npx @smithery/cli install @neondatabase/mcp-server-neon --client claude
````
```suggestion
npx -y @smithery/cli install neon --client claude
```

pffigueiredo reviewed (approved):
Looks great, will proceed with merging this into upstream now 🚀
